erpnext/accounts/doctype/payment_entry/payment_entry.py

```diff
@@ -832,9 +833,14 @@ def get_outstanding_on_journal_entry(name):
 
 @frappe.whitelist()
 def get_reference_details(reference_doctype, reference_name, party_account_currency):
-	total_amount = outstanding_amount = exchange_rate = bill_no = None
+	total_amount = outstanding_amount = exchange_rate = bill_no, posting_date = None
 	ref_doc = frappe.get_doc(reference_doctype, reference_name)
 	company_currency = ref_doc.get("company_currency") or erpnext.get_company_currency(ref_doc.company)
+	try:
```

Comment:
Not acceptable. I have said this many times, you do not wrap db queries in try except clauses! You must maintain an exhaustive map of the reference doctypes to the date fields. Every one of them has one. No shortcuts! Then set the last argument of the get_value method to the appropriate field of the doctype